ebos/eclgenericoutputblackoilmodule.cc

```diff
@@ -762,6 +795,47 @@ assignToSolution(data::Solution& sol)
         // reallocated)
         tracerConcentrations_.resize(0);
     }
+
+    // Flows
+    if (FluidSystem::phaseIsActive(gasPhaseIdx) && !flowsi_[gasCompIdx].empty())
```

Comment:
It seems to me that the first condition is unneeded: there should be no way to get a nonempty `flowsi_[gasCompIdx]` if we have no active gas phase. So only the second part of the condition (emptyness) is needed, and then the generic mechanism used on lines 699--752 (line numbers from your branch) should be used. You may have to be careful about making sure the vectors all exist (I assume they do, not read as far), but are empty, in case of a nonactive phase.